os/arch/arm/src/armv7-m/up_hardfault.c

```diff
@@ -134,17 +135,16 @@ int up_hardfault(int irq, FAR void *context, FAR void *arg)
 
 #ifdef CONFIG_BUILD_PROTECTED
 	/* In the kernel build, SVCalls are expected in either the base, kernel
-	 * FLASH region or in the user FLASH region.
+	 * FLASH region, kernel RAM region or in the user FLASH region.
 	 */
 
-	if (((uintptr_t)pc >= (uintptr_t)&_stext && (uintptr_t)pc < (uintptr_t)&_etext) ||
-			(sched_self()->uspace &&
+	if ((is_kernel_text_space((void *)pc)) || (sched_self()->uspace &&
```

Comment:
Is it wrong if we call `is_kernel_space` instead of `is_kernel_text_space`?

Reply:
I think these are a little different. 
*is_kernel_space* > Checks for text section, data & bss of kernel, whereas
*is_kernel_text_space* > Checks for only text section of kernel

Reply:
Yeah, the functions are different. But what I want to know is "do we need to use them separately in our usages?".

Reply:
If we call is_kernel_space instead of is_kernel_text_space, is there any problem?

Reply:
That is not an issue. Just to limit no. of checks that are being used in hard fault codes & for better code organization this is a better choice.